### DISCUSSION TITLE
Docs: Question

### DISCUSSION THREAD

**Discussion by datengraben** [metreeca/self#10] [General]
Hey thanks for the great tool! I use it to explore some of the endpoints I frequently use. Do you still maintain the code?

In the docs you describe how to improve user experience, especially when `self` can't find for example a label. But how can I insert the suggested SPARQL-query-snippets into the query to the server? Is there a way to do this via the user interface?

At the moment I'm figuring out how to enable `self` to harvest the relevant schemata to display them as items in the `collections`-tab on startup. Maybe you have a hint here where to look in the code?

Anyhow: Thanks for making this open source!

Best, Chris

**Comment by knoan** [comment]
https://metreeca.github.io/self/how-to/improve-user-experience#identify-key-collections

> Do you still maintain the code?

The project is currently feature-frozen and in maintenance mode: we plan to migrate most functionalities from GWT to a TypeScript/React implementation, with initial releases tentatively slated for Q4 2024.

> In the docs you describe how to improve user experience, especially when self can't find for example a label. But how can I insert the suggested SPARQL-query-snippets into the query to the server? Is there a way to do this via the user interface?

Unfortunately not: the UI is strictly read-only, as it's intended to work also on public and third-party endpoints.

> At the moment I'm figuring out how to enable self to harvest the relevant schemata to display them as items in the collections-tab on startup. Maybe you have a hint here where to look in the code?

If you have write access to the endpoint, you may fine tune the landing page with `void:rootResource` annotations as described @ https://metreeca.github.io/self/how-to/improve-user-experience#identify-key-collections; if no annotations are found, available types will be introspected using the `?resource rdf:type ?type` pattern.
